olog.py

```diff
@@ -16,77 +13,159 @@ def __init__(self, url, user, password):
-        res.raise_for_status()
-        return res.json()
+    # Logbooks
+    async def aget_logbooks(self):
+        async with self._session as api:
+            res = await api.get('logbooks')
+            res.raise_for_status()
+            return res.json()
+
+    def get_logbooks(self):
+        return asyncio.run(self.aget_logbooks())
 
-    def delete_logbook(self, name):
+    async def aget_logbook(self, name):
         # Logbooks have an integer id, but this REST endpoint expects the
         # *name*.  It does not accept an id.
-        url = f'{self._url}/resources/logbooks/{name}'
-        res = self._session.delete(url, **self._kwargs)
-        res.raise_for_status()
-        return res.json()
+        async with self._session as api:
+            res = await api.get(f'logbooks/{name}')
+            res.raise_for_status()
+            return res.json()
 
-    def post_logbook(self, logbook):
-        """
-        Create a logbook.
-        """
-        url = f'{self._url}/resources/logbooks'
-        res = self._session.post(url, data=logbook)
-        res.raise_for_status()
-        return res.json()
+    def get_logbook(self, name):
+        return asyncio.run(self.aget_logbook(name))
+
+    async def aput_logbooks(self, logbooks):
+        async with self._session as api:
+            res = await api.put('logbooks', json=logbooks)
+            res.raise_for_status()
+            return res.json()
+
+    def put_logbooks(self, logbooks):
+        return asyncio.run(self.aput_logbooks(logbooks))
+
+    async def aput_logbook(self, logbook):
+        async with self._session as api:
+            name = logbook['name']
+            res = await api.put(f'logbooks/{name}',  json=logbook)
+            res.raise_for_status()
+            return res.json()
 
     def put_logbook(self, logbook):
-        """
-        Create or update a logbook (mathced by name).
-        """
-        url = f'{self._url}/resources/logbooks/{logbook["name"]}'
-        res = self._session.put(url, data=logbook)
-        res.raise_for_status()
-        return res.json()
-
-    def list_logs(self, logbook=None, tag=None, componentType=None,
-                  start=None, end=None,
-                  search=None,
-                  **params):
-        if isinstance(start, datetime):
-            start = start.timestamp()
-        if isinstance(end, datetime):
-            end = end.timestamp()
-        params.update(dict(logbook=logbook,
-                           tag=tag,
-                           componentType=componentType,
-                           search=search,
-                           start=start,
-                           end=end))
-        url = f'{self._url}/resources/logs'
-        res = self._session.get(url, params=params, **self._kwargs)
-        res.raise_for_status()
-        return res.json()
+        return asyncio.run(self.aput_logbook(logbook))
+
+    # Logs
+    async def aget_logs(self, **params):
```

Comment:
The advantage of blindly accepting kwargs like `**params` is that it's simple and does not require updating when the underlying API changes to add/remove/rename parameters. The disadvantage is that the user has to know by some other means what parameters are allowed, and if the user uses an unexpected parameter the code fails late (in an HTTP response) rather than early (at the top of the function call).

We had to nice datetime-parsing code here as well, deleted above. Should we keep that?